libraries/AP_HAL_PX4/CAN.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2015-2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
+#include "CAN.h"
+
+extern const AP_HAL::HAL& hal;
+
+using namespace PX4;
+
+// TODO: MonotonicTime
+int16_t PX4CAN::send(const uavcan::CanFrame& frame, uavcan::MonotonicTime tx_deadline, uavcan::CanIOFlags flags)
+{
+    return 0;
+}
+
+// TODO: MonotonicTime, UtcTime
+int16_t PX4CAN::receive(uavcan::CanFrame& out_frame, uavcan::MonotonicTime& out_ts_monotonic, uavcan::UtcTime& out_ts_utc,
+        uavcan::CanIOFlags& out_flags)
+{
+    return 0;
+}
+
+int16_t PX4CAN::configureFilters(const uavcan::CanFilterConfig* filter_configs, uint16_t num_configs)
+{
+    return 0;
+}
+
+uint16_t PX4CAN::getNumFilters() const
+{
+    return 0;
+}
+
+uint64_t PX4CAN::getErrorCount() const
+{
+    return 0;
+}
+
+bool PX4CAN::begin(uint32_t bitrate)
+{
+    return 0;
+}
+
+bool PX4CAN::begin(uint32_t bitrate, float samplepoint)
+{
+    return 0;
+}
+
+void PX4CAN::end()
+{
+
+}
+
+void PX4CAN::reset()
+{
+
+}
+
+bool PX4CAN::is_initialized()
+{
+    return 0;
+}
+
+int32_t PX4CAN::tx_pending()
+{
+    return 0;
+}
+
+uint32_t PX4CAN::available()
+{
+    return 0;
+}
+
+uavcan::ICanIface* PX4CANDriver::getIface(uint8_t iface_index)
+{
+    return NULL;
```

Comment:
AFAIK, ArduPilot is being written in C++14 (or is it 11? doesn't matter), so `nullptr` should be used. (this also applies to the similar issues below)